spring-integration-ftp/src/main/java/org/springframework/integration/ftp/inbound/FtpInboundFileSynchronizer.java

```diff
@@ -42,6 +43,20 @@ public FtpInboundFileSynchronizer(SessionFactory<FTPFile> sessionFactory) {
 		super(sessionFactory);
 	}
 
+	@Override
+	public void setRemoteDirectoryExpression(Expression remoteDirectoryExpression) {
+		if (remoteDirectoryExpression != null) {
+			super.setRemoteDirectoryExpression(remoteDirectoryExpression);
+		}
+	}
+
+	@Override
+	public void afterPropertiesSet() {
+		if (getRemoteDirectoryExpression() == null) {
+			setRemoteDirectory(null);
+		}
+		super.afterPropertiesSet();
+	}
```

Comment:
Because the `AbstractInboundFileSynchronizer` is extended by the SFTP, too, which doesn't allow `null` for remote dir.